source/ring_buffer.c

```diff
@@ -73,7 +73,8 @@ int aws_ring_buffer_acquire(struct aws_ring_buffer *ring_buf, size_t requested_s
 
     /* this branch is, we don't have any vended buffers. */
     if (head_cpy == tail_cpy) {
-        size_t ring_space = ring_buf->allocation_end - ring_buf->allocation;
+        size_t ring_space =
+            (!ring_buf->allocation || !ring_buf->allocation_end) ? 0 : ring_buf->allocation_end - ring_buf->allocation;
```

Comment:
Is there a reason why we check both  `ring_buf->allocation` and `ring_buf->allocation_end` now?

Why not:

```c
size_t ring_space = ring_buf->allocation_end == NULL ? 0 : ring_buf->allocation_end - ring_buf->allocation;
```

which is more specific to the issue this PR is addressing.